src/main/java/com/minecolonies/coremod/colony/jobs/AbstractJob.java

```diff
@@ -296,7 +296,8 @@ public ItemStack removeItemNeeded(@NotNull final ItemStack stack)
                 neededItem.stackSize -= itemsToRemove;
                 stackCopy.stackSize -= itemsToRemove;
 
-                if (neededItem.stackSize == 0)
+                //Deativate this for now in order to keep working even if not all items are given.
+                //if (neededItem.stackSize == 0)
                 {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this nested code block into a method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1199)